ql/src/java/org/apache/hadoop/hive/ql/io/avro/AvroGenericRecordReader.java

```diff
@@ -88,22 +88,39 @@ public AvroGenericRecordReader(JobConf job, FileSplit split, Reporter reporter)
       gdr.setExpected(latest);
     }
 
-    if (split.getLength() == 0) {
-      this.isEmptyInput = true;
-      this.start = 0;
-      this.reader = null;
+    org.apache.avro.file.FileReader<GenericRecord> nonFinalReader = null;
+    try {
```

Comment:
Can we put this `try` to the `else` only?
Maybe even try with resources if the reader is autoclosable, like
```
try (org.apache.avro.file.FileReader<GenericRecord> nonFinalReader = new DataFileReader<GenericRecord>(new FsInput(split.getPath(), job), gdr)) {
[..]
}
```

Reply:
Can't do. See comment above. Logically I am operating here on the `AvroGenericRecordReader.reader`. I just need a temporary `nonFinalReader` instance variable so I can access the field in `catch` clause. Otherwise compiler would complain that field may not be initialized yet.

Reply:
Got it, thanks!